nixos/modules/services/misc/klipper.nix

```diff
@@ -69,6 +67,26 @@ in
           for supported values.
         '';
       };
+
+      firmware = mkOption {
+        description = "Firmwares klipper should manage";
+        default = { };
+        type = with types; attrsOf
+          (submodule {
+            options = {
+              enableFlashing = mkEnableOption ''
+                automatic flashing of firmware to microcontroller
+
+                WARNING: Be careful with this option. Enabling this will automatically flash your microcontroller on e.g. nixos-rebuild.
+                This can potentially brick your microcontroller. Alternatively, use `klipper-flash-$mcu` to flash manually.
+              '';
+              firmwareConfig = mkOption {
+                type = path;
+                description = "Path to firmware config which is generated using `klipper-genconf`";
```

Comment:
I would prefer is the user could just pass a string and we called `klipper-genconf` with `runCommand`.

Reply:
I think there is a misunderstanding in what `klipper-genconf` is intended to do:

The current flow to get the `configFile` for the firmware is as following:
1. Undestand how klipper builds the firmware
2. Clone klipper repo
3. `nix-shell -p python2`
4. `make menuconfig`
5. Look for the `.config` file

`klipper-genconf` compresses all of them for convenience. It is intended as a standalone helper script which is used once to retrieve the firmware configuration that klipper understands. Internally it uses menuconfig, something which is interactive. So with this tool, hopefully nobody has to understand how the build process with all the Makefiles at klipper works and can just generate and use it.